test/hotspot/jtreg/containers/systemd/SystemdMemoryAwarenessTest.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2024, Red Hat, Inc.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+
+/*
+ * @test
+ * @summary Memory/CPU awareness test for JDK-under-test inside a systemd slice.
+ * @requires systemd.support
+ * @library /test/lib
+ * @build HelloSystemd jdk.test.whitebox.WhiteBox
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller -jar whitebox.jar jdk.test.whitebox.WhiteBox
+ * @run main/othervm -Xbootclasspath/a:whitebox.jar -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI SystemdMemoryAwarenessTest
+ */
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import jdk.test.lib.Platform;
+import jdk.test.whitebox.WhiteBox;
+import jdk.test.lib.containers.systemd.SystemdRunOptions;
+import jdk.test.lib.containers.systemd.SystemdTestUtils;
+import jdk.test.lib.containers.systemd.SystemdTestUtils.ResultFiles;
+
+import jtreg.SkippedException;
+
+public class SystemdMemoryAwarenessTest {
+
+    private static final WhiteBox wb = WhiteBox.getWhiteBox();
+
+    public static void main(String[] args) throws Exception {
+       if (!Platform.isRoot()) {
+           throw new SkippedException("Test requires to be run as root");
+       }
+       testHelloSystemd();
+    }
+
+    private static void testHelloSystemd() throws Exception {
+        SystemdRunOptions opts = SystemdTestUtils.newOpts("HelloSystemd");
+        // 1 GB memory
+        opts.memoryLimit("1000M");
```

Comment:
Just wondering - is 1G here possible (the comment states 1 GB / 1024M) ?

Reply:
I probably shall fix the comment or change it to `1024M`. Either way it has to match the assertion where we look for `1048576000` bytes in the output.

Reply:
This should be fixed now.